base/slicearray.jl

```diff
@@ -232,3 +259,37 @@ Base.@propagate_inbounds setindex!(s::Slices{P,SM,AX,S,N}, val, I::Vararg{Int,N}
     s.parent[_slice_index(s, I...)...] = val
 
 parent(s::Slices) = s.parent
+
+function _element_size(s::Slices)
+    long = map((n,l) -> l === (:) ? n : nothing, size(s.parent), s.slicemap)
+    filter(!isnothing, long)
+end
+
+# These control summary printing, like `3-element eachcol(adjoint(::Matrix{Int64})) of ...`
+function showarg(io::IO, s::ColumnSlices, toplevel)
+    print(io, "eachcol(")
+    showarg(io, parent(s), false)
+    print(io, ')')
+    toplevel && print(io, " of ", dims2string(_element_size(s)), " slices with eltype ", eltype(eltype(s)))
+    return nothing
+end
+function showarg(io::IO, s::RowSlices, toplevel)
+    print(io, "eachrow(")
+    showarg(io, parent(s), false)
+    print(io, ')')
+    toplevel && print(io, " of ", dims2string(_element_size(s)), " slices with eltype ", eltype(eltype(s)))
+    return nothing
+end
+function showarg(io::IO, s::Slices, toplevel)
+    drop = ndims(s) + ndims(eltype(s)) > ndims(parent(s))
+    dims_vec = filter(c -> c isa Integer, [findfirst(==(d), s.slicemap) for d in 1:ndims(parent(s))])
+    dims = length(dims_vec) == 1 ? only(dims_vec) : Tuple(dims_vec)
+    print(io, "eachslice(")
+    showarg(io, parent(s), false)
+    print(io, ", dims = ", dims, drop ? ", drop = false)" : ")")
+    toplevel && print(io, " of ", dims2string(_element_size(s)), " slices with eltype ", eltype(eltype(s)))
+    return nothing
+end
+
+# This hides the SubArray type in compact printing, e.g. (0, eachcol(Float32[1 2; 3 4]), 5)
```

Comment:
I don't really understand this comment. Maybe develop a bit more?

Reply:
To be clear what this is doing, this is the effect:
```julia
julia> (0, eachcol(Float32[1 2; 3 4]), 5)  # master
(0, SubArray{Float32, 1, Matrix{Float32}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}[[1.0, 3.0], [2.0, 4.0]], 5)

julia> (0, eachcol(Float32[1 2; 3 4]), 5)  # this PR
(0, [Float32[1.0, 3.0], Float32[2.0, 4.0]], 5)
```
The default 2-arg show for an array of arrays prints the whole type of the inner arrays as the eltype of the outer. This line removes that. But also notifies `show` that the eltype of the inner arrays has not been printed anywhere, hence they become `Float32[...]`. 

Will expand the comment a bit.

Reply:
Expanded in [f37584d](https://github.com/JuliaLang/julia/pull/45355/commits/f37584d022e4b58998f4cc1ffc5e79778dd38306)

Reply:
Yes, I'm not sure. Compact printing is sometimes lossy, e.g. Transpose is not marked here (but matters):
```
julia> ([1 2], transpose([3,4]), [[5,6], [6,7]]')
([1 2], [3 4], Adjoint{Int64, Vector{Int64}}[[5 6] [6 7]])

julia> Adjoint{Int64, Vector{Int64}}[[5 6] [6 7]]
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type Adjoint{Int64, Vector{Int64}}
```
Also, the Adjoint printing is not copy-pastable, it does not re-create such elements. That's also true of the SubArray types which will be shown for eachcol. Perhaps both should be abbreviated somehow?

Reply:
[f3869c9](https://github.com/JuliaLang/julia/pull/45355/commits/f3869c9565933e54bc6210d6d68a48f16bf287c7) reverts this for now.